src/psyclone/tests/gocean1p0_build.py

```diff
@@ -0,0 +1,139 @@
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors: J. Henrichs, Bureau of Meteorology
+
+
+''' Module containing configuration required to build code generated
+for the Dynamo0p3 API '''
+
+from __future__ import absolute_import, print_function
+import os
+from psyclone_test_utils import Compile, CompileError
+
+
+class GOcean1p0Build(Compile):
+    '''Build class for compiling test files for the GOcean1.0 api. It
+    uses dl_esm_inf as included in the PSyclone distribution.
+    '''
+
+    _infrastructure_built = False
+    _infrastructure_path = ""
+
+    def __init__(self, tmpdir):
+        super(GOcean1p0Build, self).__init__(tmpdir)
+
+        base_path = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                                 "test_files", "gocean1p0")
+        self.base_path = base_path
+
+        # On first instantiation (triggered by conftest.infra_compile)
+        # compile the infrastructure library files.
+        if Compile.TEST_COMPILE and not GOcean1p0Build._infrastructure_built:
+            self._build_infrastructure()
+            GOcean1p0Build._infrastructure_path = str(tmpdir)
+
+    def get_infrastructure_flags(self):
+        '''Returns the required flag to use the infrastructure library dl_esm_inf
+        for gocean1p0. Each parameter must be a separate entry
+        in the list, e.g.: ["-I", "/some/path"] and not ["-I /some/path"].
+        :returns: A list of strings with the compiler flags required.
+        :rtpe: list
+        '''
+        return ["-I", self._infrastructure_path]
+
+    def _build_infrastructure(self):
+        '''Compile dl_esm_inf.
+        '''
+
+        old_pwd = self._tmpdir.chdir()
+
+        import subprocess
+        dl_esm_inf_path = \
+            os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                         "../../../external/dl_esm_inf/finite_difference/src")
+
+        arg_list = ["make", "F90={0}".format(self._f90),
+                    "F90FLAGS={0}".format(self._f90flags),
+                    "-f", "{0}/Makefile".format(dl_esm_inf_path)]
+        try:
```

Comment:
The remainder of this method looks very similar to the equivalent in the Dynamo0p3 version of this class. Could the running of the compilation itself and the following error checking be moved into the base class? However, if that would be complicated then don't worry about it.

Reply:
The code in Gocean1p0Build._build_infrastructure shares some commonality with Compile._compile_file, not with Dynamo0p3Build - hmm - well, maybe a little bit, catching errors, but I don't see how this could easily be done in one base class (gocean calls popen with build, dynamo calls compile_file in a loop). Maybe a ``print_error`` function or so depending on the exception, but this would likely apply to a lot of other code as well, way beyond the scope of this pull request. So tbh, I'd say for now to just leave it the way it is.

Reply:
OK.